In [16]:
# download all pictures based off of the url scraped from Overstock
# csv file in working directory. File was named image_url_061119update6th.csv
import os
import time
import pandas as pd
import csv
import requests

# store working directory
dir = os.getcwd()

In [12]:
img_url = pd.read_csv("./image_url_061119update6th.csv",index_col = 2)
print(img_url.head())
print(img_url.shape)

          Unnamed: 0                                                url
index                                                                  
Page_1_1           0  https://ak1.ostkcdn.com/images/products/941290...
Page_1_2           1  https://ak1.ostkcdn.com/images/products/132185...
Page_1_3           2  https://ak1.ostkcdn.com/images/products/881687...
Page_1_4           3  https://ak1.ostkcdn.com/images/products/662457...
Page_1_5           4  https://ak1.ostkcdn.com/images/products/881715...
(1879, 2)


In [13]:
# remove the first column from the dataframe
img_url.drop(columns="Unnamed: 0", inplace = True)
print(img_url.head())

                                                        url
index                                                      
Page_1_1  https://ak1.ostkcdn.com/images/products/941290...
Page_1_2  https://ak1.ostkcdn.com/images/products/132185...
Page_1_3  https://ak1.ostkcdn.com/images/products/881687...
Page_1_4  https://ak1.ostkcdn.com/images/products/662457...
Page_1_5  https://ak1.ostkcdn.com/images/products/881715...


In [17]:
test = img_url.iloc[0,0]
print(test)
print(type(test))
print(img_url.index.values[0])

https://ak1.ostkcdn.com/images/products/9412909/L16600539.jpg?imwidth=480&impolicy=medium
<class 'str'>
Page_1_1


In [22]:
for i in range(5):
    current_url = img_url.iloc[i,0]
    img_name = img_url.index.values[i]
    print([current_url , img_name])

['https://ak1.ostkcdn.com/images/products/9412909/L16600539.jpg?imwidth=480&impolicy=medium', 'Page_1_1']
['https://ak1.ostkcdn.com/images/products/13218518/L19936654.jpg?imwidth=480&impolicy=medium', 'Page_1_2']
['https://ak1.ostkcdn.com/images/products/8816874/L16050950.jpg?imwidth=480&impolicy=medium', 'Page_1_3']
['https://ak1.ostkcdn.com/images/products/6624571/L14191181.jpg?imwidth=480&impolicy=medium', 'Page_1_4']
['https://ak1.ostkcdn.com/images/products/8817153/L16051170.jpg?imwidth=480&impolicy=medium', 'Page_1_5']


In [31]:
# credit to: https://www.quora.com/How-do-I-write-a-Python-code-to-download-images-from-100-URLs-stored-in-a-CSV-file
for i in range(img_url.shape[0]):
    current_url = img_url.iloc[i,0]
    img_name = img_url.index.values[i] + '.jpg'
    result = requests.get (current_url, stream = True)
    if result.status_code == 200:
        image = result.raw.read()
        open(os.path.join(dir,r'Pillow_img',img_name),'wb').write(image)
        


In [24]:
print(img_url.shape[0])

1879


Downloaded 1879 pillow images in 480x480 size to folder "Pillow_img"